### PR TITLE
TEST: adopt new rest catalog image and enable tableExists tests

### PR DESCRIPTION
**Pull request by sungwy**
NOTE: the test is failing because tableExists endpoint isn't exposed on the adapter. Related fix: https://github.com/apache/iceberg/pull/11678

### CODE REVIEW

File: dev/docker-compose-integration.yml [hunk 1]
```diff
@@ -41,7 +41,8 @@ services:
       - hive:hive
       - minio:minio
   rest:
-    image: tabulario/iceberg-rest
+    # TODO: This needs to be updated to apache/iceberg-rest-fixture once https://github.com/apache/iceberg/pull/11676 is merged
+    image: asfjenkins/iceberg-rest-fixture
```
Comment (Fokko) [inline dev/docker-compose-integration.yml:45]:
```suggestion
    image: apache/iceberg-rest-fixture
```

**Comment by Fokko** [comment]
The image is published, including the fix around the `tableExists`, so 🤞

**Comment by Fokko** [comment]
Hmm, still an issue with the `tableExists`:

```
    @pytest.mark.integration
    def test_table_v1_with_null_nested_namespace(session_catalog: Catalog, arrow_table_with_null: pa.Table) -> None:
        identifier = "default.lower.table_v1_with_null_nested_namespace"
        tbl = _create_table(session_catalog, identifier, {"format-version": "1"}, [arrow_table_with_null])
        assert tbl.format_version == 1, f"Expected v1, got: v{tbl.format_version}"
    
        assert session_catalog.load_table(identifier) is not None
>       assert session_catalog.table_exists(identifier)
E       AssertionError: assert False
E        +  where False = <bound method RestCatalog.table_exists of local (<class 'pyiceberg.catalog.rest.RestCatalog'>)>('default.lower.table_v1_with_null_nested_namespace')
E        +    where <bound method RestCatalog.table_exists of local (<class 'pyiceberg.catalog.rest.RestCatalog'>)> = local (<class 'pyiceberg.catalog.rest.RestCatalog'>).table_exists

tests/integration/test_writes/test_writes.py:1373: AssertionError
=============================== warnings summary ========================
```

**Comment by sungwy** [comment]
:) @Fokko I'm on my way to the office, but will take a look at this in a few minutes

**Comment by sungwy** [comment]
@Fokko - it looks like the issue is because the rest-fixture is returning a 200 response. I've put in edits to allow PyIceberg to infer a 200 response as being successful. 

The proposed solution deviates from the [REST Catalog Spec](https://github.com/apache/iceberg/blob/c7cef9bd81d150108a56a848b7482f179ea878b8/open-api/rest-catalog-open-api.yaml#L1691-L1699), but this is already the behavior of PyIceberg when handles the success codes of the other endpoints that are also meant to return 204 response:

https://github.com/apache/iceberg-python/blob/df804c4c65033039ef16bb9a51c68aebfb010e5e/pyiceberg/catalog/rest.py#L422-L448

**Comment by Fokko** [comment]
Thanks for following this up @sungwy. I think it is okay to mark all 2xx responses as table exists. Polaris had a similar issue: https://github.com/apache/iceberg-python/issues/1363

**Review by Fokko — APPROVED** [review]

**Review by kevinjqliu — APPROVED** [review]
woot! LGTM

**Comment by sungwy** [comment]
Thanks for the reviews @Fokko and @kevinjqliu !

**Pull request merged**